modules/js-and-css/speculation-rules/settings.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * Settings functions used for Speculation Rules.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Returns the available options for the Speculation Rules mode and their labels.
+ *
+ * @since n.e.x.t
+ *
+ * @return array Associative array of `$mode => $label` pairs.
+ */
+function plsr_get_mode_labels() {
+	return array(
+		'prefetch'  => _x( 'Prefetch', 'setting label', 'performance-lab' ),
+		'prerender' => _x( 'Prerender', 'setting label', 'performance-lab' ),
+	);
+}
+
+/**
+ * Returns the available options for the Speculation Rules eagerness and their labels.
+ *
+ * @since n.e.x.t
+ *
+ * @return array Associative array of `$eagerness => $label` pairs.
+ */
+function plsr_get_eagerness_labels() {
+	return array(
+		'conservative' => _x( 'Conservative (on click)', 'setting label', 'performance-lab' ),
+		'moderate'     => _x( 'Moderate (on hover)', 'setting label', 'performance-lab' ),
```

Comment:
These are the heuristics that are landed in Chrome today. We might tweak/improve these to not be exactly this in the future (for example, "hover" isn't really a thing on mobile but we're looking at other heuristics there). And other browsers might use other heuristics especially on platforms with other kinds of things that indicate an upcoming click (e.g., if the user generally selects links with the keyboard rather than a pointing device).

It might still be reasonable to describe in this way to get an intution, but that's the full context here.

Reply:
Thanks! Good point, hover doesn't make sense for mobile. Still, I feel this may be the shortest way to describe it to give people a better idea.

Reply:
Updated in https://github.com/WordPress/performance/pull/939/commits/9a6c3a99cdeeebc0277769cdf3ba9e607ac2b39d

Reply:
I think the use of "typically" here is good. The below sentence,

> Whether to trigger on click (conservative), on hover (moderate), or on even the slight suggestion (eager) that the user may navigate to the URL.

seems more problematic, but maybe adding "typically" could help there too?

Or you could try replacing that sentence with a more vague one, e.g. below you have

> The eagerness setting defines the heuristics based on which the loading is triggered.

and

> "Eager" will have the minimum delay to start speculative loads, "Conservative" increases the chance that only URLs the user actually navigates to are loaded.

which are both quite good IMO.

Reply:
Thanks, updated in https://github.com/WordPress/performance/pull/939/commits/da0b883ba57a08456019aa21289e453cbb6173c7